### ISSUE TITLE
Reduce usage of nighlty features

### ISSUE THREAD

**Issue by ZetaNumbers** [overdrivenpotato/rust-psp#156]
`psp/src/panic.rs` breaks a lot lately, maybe gate this under a cargo feature and use `panic = "abort"` by default.

**Comment by SK83RJOSH** [comment]
This issue should probably be renamed, the issue with panics seems to be some sort of linker change upstream in `rust-lld`. My general suggestion would be to move to `unwinding` and figure out what workaround mitigates the underlying issue/isolate what change in `rust-lld` broke them in the first place. 

In lieu of that, panic support should be dropped outright, though I think that would be rather unfortunate.

**Comment by zetanumbers** [comment]
> This issue should probably be renamed…

No, don't think it's just about the panics. In general we would want this to reduce severity of future breakages. If we want to focus on a specific aspect, we could open a new issue or PR for that.

**Comment by SK83RJOSH** [comment]
> In general we would want this to reduce severity of future breakages.

Fair enough, though I suppose an effort should be made to enumerate the most fragile things. Ie/ internal features dependencies, and so on. 

But yeah I'll probably spin off a separate issue about panics, tried to get discussion going about it on discord, but it seems it's dead in the water for now. :slightly_smiling_face:
